x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/AutoscalingIT.java

```diff
@@ -0,0 +1,194 @@
+            AutoscalingDeciderResult.Reason.class,
+            MlScalingReason.NAME,
+            MlScalingReason::new
+        ));
+        return entries;
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Arrays.asList(
+            LocalStateMachineLearning.class,
+            Netty4Plugin.class,
+            ReindexPlugin.class,
+            Autoscaling.class,
+            // The monitoring plugin requires script and gsub processors to be loaded
+            IngestCommonPlugin.class,
+            // The monitoring plugin script processor references painless. Include this for script compilation.
+            // This is to reduce log spam
+            MockPainlessScriptEngine.TestPlugin.class,
+            // ILM is required for .ml-state template index settings
+            IndexLifecycle.class,
+            DataStreamsPlugin.class);
+    }
+
+    @Override
+    protected Settings externalClusterClientSettings() {
+        return Settings.builder().put(super.externalClusterClientSettings())
+            .put(Autoscaling.AUTOSCALING_ENABLED_SETTING.getKey(), true)
+            .build();
+    }
+
+    public void testMLAutoscalingCapacity() {
+        SortedMap<String, Settings> deciders = new TreeMap<>();
+        deciders.put(MlAutoscalingDeciderService.NAME, Settings.builder().build());
+        final PutAutoscalingPolicyAction.Request request = new PutAutoscalingPolicyAction.Request(
+            "ml_test",
+            new TreeSet<>(Arrays.asList("ml")),
+            deciders
+        );
+        assertAcked(client().execute(PutAutoscalingPolicyAction.INSTANCE, request).actionGet());
+
+        assertMlCapacity(
+            client().execute(
+                GetAutoscalingCapacityAction.INSTANCE,
+                new GetAutoscalingCapacityAction.Request()
+            ).actionGet(),
+            "Requesting scale down as tier and/or node size could be smaller",
+            0L,
+            0L);
+
+        putJob("job1", 100);
+        putJob("job2", 200);
+        openJob("job1");
+        openJob("job2");
+
+        assertMlCapacity(
+            client().execute(
+                GetAutoscalingCapacityAction.INSTANCE,
+                new GetAutoscalingCapacityAction.Request()
+            ).actionGet(),
+            "Requesting scale down as tier and/or node size could be smaller",
+            1328196267L,
+            734003200L);
+
+        putJob("bigjob1", 20_000);
+        putJob("bigjob2", 10_000);
+        putJob("bigjob3", 30_000);
+        openJob("bigjob1");
+        openJob("bigjob2");
+        openJob("bigjob3");
+
+        assertMlCapacity(
+            client().execute(
+                GetAutoscalingCapacityAction.INSTANCE,
+                new GetAutoscalingCapacityAction.Request()
+            ).actionGet(),
+            "requesting scale up as number of jobs in queues exceeded configured limit",
+            380991001934L,
```

Comment:
I can see where the `1328196267L` comes from in the previous assertion: `ceil((100 + 40 + 200 + 40) * 1024^2 * 100 / 30)`.

But why is this one `380991001934L`?  `ceil((100 + 40 + 200 + 40 + 20000 + 40 + 10000 + 40 + 30000 + 40) * 1024^2 * 100 / 30) = 211462826667`

I think it would be good to put the expected formula that's been evaluated in a comment so that if ever there are changes to the constants used in the code then the person who edits the expected result here doesn't just mindlessly paste in whatever makes the test pass but can use the formula to manually check the result makes sense given their new constants.